File: python/best-time-buy-and-sell-stock-ii.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        res = 0
        curr = prices[0]
        
        for i in range(1, len(prices)):
            if prices[i] < curr:
                curr = prices[i]
            else:
                res += prices[i] - curr
                curr = prices[i]
        return res

File: go/best_time_buy_and_sell_stock_ii.go
package leetcode

func maxProfit(prices []int) int {
	res, curr := 0, prices[0]

	for i := 1; i < len(prices); i++ {
		if prices[i] < curr {
			curr = prices[i]
		} else {
			res += prices[i] - curr
			curr = prices[i]
		}
	}
	return res
}
